tests/common/test_common_models.py

```diff
@@ -27,3 +37,240 @@ def test_calendar_timestamps():
+    )
+
+    assert isinstance(order.id, UUID)
+    assert isinstance(order.client_order_id, UUID)
+    assert isinstance(order.replaced_by, UUID)
+    assert isinstance(order.replaces, UUID)
+    assert isinstance(order.asset_id, UUID)
+
+
+def test_order_legs():
+    """Tests recursive Order object with legs field"""
+
+    order = Order(
+        id="61e69015-8549-4bfd-b9c3-01e75843f47d",
+        client_order_id="eb9e2aaa-f71a-4f51-b5b4-52a6c565dad4",
+        created_at="2021-03-16T18:38:01.942282Z",
+        updated_at="2021-03-16T18:38:01.942282Z",
+        submitted_at="2021-03-16T18:38:01.937734Z",
+        filled_at="2021-03-16T18:38:01.937734Z",
+        expired_at="2021-03-16T18:38:01.937734Z",
+        canceled_at="2021-03-16T18:38:01.937734Z",
+        failed_at="2021-03-16T18:38:01.937734Z",
+        replaced_at="2021-03-16T18:38:01.937734Z",
+        replaced_by="61e69015-8549-4bfd-b9c3-01e75843f47d",
+        replaces="61e69015-8549-4bfd-b9c3-01e75843f47d",
+        asset_id="b0b6dd9d-8b9b-48a9-ba46-b9d54906e415",
+        symbol="AAPL",
+        asset_class="us_equity",
+        notional="500",
+        qty=None,
+        filled_qty="0",
+        filled_avg_price=None,
+        order_class="",
+        order_type="market",
+        type="market",
+        side="buy",
+        time_in_force="day",
+        limit_price=None,
+        stop_price=None,
+        status="accepted",
+        extended_hours=False,
+        legs=None,
+        trail_percent=None,
+        trail_price=None,
+        hwm=None,
+        commission="1.25",
+    )
+
+    order_with_legs = Order(
+        id="61e69015-8549-4bfd-b9c3-01e75843f47d",
+        client_order_id="eb9e2aaa-f71a-4f51-b5b4-52a6c565dad4",
+        created_at="2021-03-16T18:38:01.942282Z",
+        updated_at="2021-03-16T18:38:01.942282Z",
+        submitted_at="2021-03-16T18:38:01.937734Z",
+        filled_at="2021-03-16T18:38:01.937734Z",
+        expired_at="2021-03-16T18:38:01.937734Z",
+        canceled_at="2021-03-16T18:38:01.937734Z",
+        failed_at="2021-03-16T18:38:01.937734Z",
+        replaced_at="2021-03-16T18:38:01.937734Z",
+        replaced_by="61e69015-8549-4bfd-b9c3-01e75843f47d",
+        replaces="61e69015-8549-4bfd-b9c3-01e75843f47d",
+        asset_id="b0b6dd9d-8b9b-48a9-ba46-b9d54906e415",
+        symbol="AAPL",
+        asset_class="us_equity",
+        notional="500",
+        qty=None,
+        filled_qty="0",
+        filled_avg_price=None,
+        order_class="",
+        order_type="market",
+        type="market",
+        side="buy",
+        time_in_force="day",
+        limit_price=None,
+        stop_price=None,
+        status="accepted",
+        extended_hours=False,
+        legs=[order],
+        trail_percent=None,
+        trail_price=None,
+        hwm=None,
+        commission="1.25",
+    )
+
+    assert isinstance(order_with_legs.legs, list)
+    assert isinstance(order_with_legs.legs[0], Order)
+
+
+class TestClosePositionRequest:
```

Comment:
Why did you make this a test class? If possible can we please keep the tests as free functions ?

Reply:
Fixed.